public/app/overrides/services/apps.ts

```diff
@@ -0,0 +1,87 @@
+import { App, AppSchema } from '@webapp/models/app';
+import { Result } from '@webapp/util/fp';
+import { z, ZodError } from 'zod';
+import type { RequestError } from '@webapp/services/base';
+import { parseResponse, requestWithOrgID } from '@webapp/services/base';
+
+type SeriesResponse = {
+  labelsSet: [
+    {
+      labels: Array<{ name: string; value: string }>;
+    }
+  ];
+};
+
+const SeriesResponseSchema = z.object({
+  labelsSet: z.array(
+    z.object({
+      labels: z.array(
+        z.object({
+          name: z.string(),
+          value: z.string(),
+        })
+      ),
+    })
+  ),
+});
+
+z.string()
+  .transform((val) => val.length)
+  .pipe(z.number().min(5));
+
+const AppsSchema = SeriesResponseSchema.transform((v) => {
+  return groupByAppAndProfileId(v);
+}).pipe(z.array(AppSchema));
+
+// TODO: change after https://github.com/grafana/phlare/pull/710 is merged
+const appTag = 'pyroscope_app';
+
+function mergeLabels(v: SeriesResponse['labelsSet'][number]['labels']) {
+  return v.reduce((acc, curr) => {
+    // TODO: type
+    acc[curr.name] = curr.value;
+    return acc;
+  }, {});
+}
+
+function addQueryAndName(v: ReturnType<typeof mergeLabels>) {
+  // TODO: type
+  const query = `${v['__profile_type__']}{${appTag}="${v[appTag]}"}`;
+
+  return {
+    ...v,
+    query,
+    FOO: 'foo',
+    name: `${v[appTag]}.${v['__profile_type__']}`, // the app selector expects this
+  };
+}
+
+//export function groupByAppAndProfileId(props: SeriesResponse) {
+export function groupByAppAndProfileId(
+  props: z.infer<typeof SeriesResponseSchema>
+) {
+  return props.labelsSet
+    .flatMap((v) => mergeLabels(v.labels))
+    .map(addQueryAndName);
+}
+
+export async function fetchApps(): Promise<
+  Result<App[], RequestError | ZodError>
+> {
+  // TODO: is this the best query?
+  const response = await requestWithOrgID('/querier.v1.QuerierService/Series', {
+    method: 'POST',
+    body: JSON.stringify({
+      matchers: [],
+    }),
+    headers: {
+      'content-type': 'application/json',
+    },
+  });
```

Comment:
@simonswine I wonder if there isn't a better way to get this data...

Reply:
Not quite, more like if we shouldn't create an endpoint that does most of this work (create an "App" that combines most of its labels). Plus I don't know how bad it currently is performance wise.

Reply:
I am unsure what the method is supposed to achieve.

* Do you just want to get a flat list of unique values for `pyroscope_app` (or soon `__service_name__`): Then you should rather use LabelValues: 
  ```curl -vL -d '{"name":"__service_name__"}' -X POST -H "content-type: application/json" http://localhost:4100/querier.v1.QuerierService/LabelValues```
  It also supports matchers just like Series
* Or do you need the full label set?: Then series is the way to go

Reply:
I think I need the full label set, since I need `__service_name__` + `profileID` (to be able to query) + other metadata fields, like `__type__` and `__name__` (to be able to display more info about what you are selecting) :(